rest/blip_api_test.go

```diff
@@ -284,7 +278,67 @@ func TestProposedChangesNoConflictsMode(t *testing.T) {
 
 }
 
+// Connect to public port with authentication
+func TestPublicPortAuthentication(t *testing.T) {
+
+	// Create bliptester that is connected as user1, with access to the user1 channel
+	btUser1 := NewBlipTesterFromSpec(BlipTesterSpec{
+		noAdminParty:       true,
+		connectingUsername: "user1",
+		connectingPassword: "1234",
+	})
+	defer btUser1.Close()
+
+	// Send the user1 doc
+	btUser1.SendRev(
+		"foo",
+		"1-abc",
+		[]byte(`{"key": "val", "channels": ["user1"]}`),
+	)
+
+	// Create bliptester that is connected as user2, with access to the * channel
+	btUser2 := NewBlipTesterFromSpec(BlipTesterSpec{
+		noAdminParty:                true,
+		connectingUsername:          "user2",
+		connectingPassword:          "1234",
+		connectingUserChannelGrants: []string{"*"},      // user2 has access to all channels
+		restTester:                  btUser1.restTester, // re-use rest tester, otherwise it will create a new underlying bucket in walrus case
+	})
+	defer btUser2.Close()
+
+	// Send the user2 doc, which is in a "random" channel, but it should be accessible due to * channel access
+	btUser2.SendRev(
+		"foo2",
+		"1-abcd",
+		[]byte(`{"key": "val", "channels": ["NBC"]}`),
+	)
+
+	// Assert that user1 received a single expected change
+	changesChannelUser1 := btUser1.GetChanges()
+	assert.True(t, len(changesChannelUser1) == 1)
+	change := changesChannelUser1[0]
+	assert.True(t, strings.Contains(change[1].(string), "foo"))
+	assert.True(t, strings.Contains(change[2].(string), "1-abc"))
+
+	// Assert that user2 received user1's change as well as it's own change
+	changesChannelUser2 := btUser2.GetChanges()
+	assert.True(t, len(changesChannelUser2) == 2)
+	change = changesChannelUser2[0]
+	assert.True(t, strings.Contains(change[1].(string), "foo"))
+	assert.True(t, strings.Contains(change[2].(string), "1-abc"))
+	change = changesChannelUser2[1]
+	assert.True(t, strings.Contains(change[1].(string), "foo2"))
+	assert.True(t, strings.Contains(change[2].(string), "1-abcd"))
```

Comment:
Should we have a helper method for assert.ChangeEquals() here?  Are we intentionally not checking sequence/deleted on the returned changes?

Reply:
Yeah great suggestion.  I added that in https://github.com/couchbase/sync_gateway/pull/3217/commits/abdbb1a39b0f669826800379226cfaae7b300e06, but did run into a caveat when type asserting the sequence into a string -- it was a float, which seems wrong and requires more investigation.  Also I think it's going to get pretty messy if we don't add a struct here and stick with the slice of empty interfaces, so maybe both of those can be tackled at the same time.   It feels like it should be in a separate PR though.

Reply:
Sure - that makes sense as part of the separate enhancement to changes struct handling you mentioned above.

Reply:
Ok, I filed https://github.com/couchbase/sync_gateway/issues/3219 for tracking that